pkg/crc/segment/segment_test.go

```diff
@@ -84,7 +87,7 @@ func TestClientUploadWithConsent(t *testing.T) {
 	c, err := newCustomClient(config, filepath.Join(dir, "telemetry"), server.URL)
 	require.NoError(t, err)
 
-	require.NoError(t, c.Upload(context.Background(), "start", time.Minute, errors.New("an error occurred")))
+	require.NoError(t, c.Upload(context.Background(), "start", time.Minute, crcErr.ToSerializableError(crcErr.VMNotExist)))
```

Comment:
Might be worth testing both cases, error type implementing `Unwrap()`, and error type not implementing `Unwrap()`